File: frontend-web/lib/mock-utils.ts
export function mockDelay<T>(data: T, ms = 600): Promise<T> {
  return new Promise((resolve) => setTimeout(() => resolve(data), ms));
}

export function mockError(message: string, ms = 600): Promise<never> {
  return new Promise((_, reject) => setTimeout(() => reject(new Error(message)), ms));
}

File: mobile_flutter/lib/data/mock/mock_utils.dart
/// Dart port of `frontend-web/lib/mock-utils.ts`. Same 600ms default delay,
/// same shape — every mock repository method awaits [mockDelay] (or throws
/// via [mockError]) instead of returning synchronously, so UI loading states
/// get exercised even against fake data, same as the web mocks.
Future<T> mockDelay<T>(T Function() value, {Duration duration = const Duration(milliseconds: 600)}) async {
  await Future.delayed(duration);
  return value();
}

Future<Never> mockError(String message, {Duration duration = const Duration(milliseconds: 600)}) async {
  await Future.delayed(duration);
  throw Exception(message);
}
